types/artifacts.d.ts

```diff
@@ -182,6 +184,26 @@ declare global {
         params: {name: string; value: string}[];
       }
 
+      export interface IFrameElement {
+        /** The `id` attribute of the iframe. */
+        id: string,
+        /** The `src` attribute of the iframe. */
+        src: string,
+        /** The iframe's ClientRect. @see https://developer.mozilla.org/en-US/docs/Web/API/Element/getBoundingClientRect */
+        clientRect: {
+          top: number;
+          bottom: number;
+          left: number;
+          right: number;
+          width: number;
+          height: number;
+        },
+        /** If the iframe or an ancestor of the iframe is fixed in position. */
+        isPositionFixed: boolean,
+        /** The Frame of the iframe, undefined if cross-origin. @see https://chromedevtools.github.io/devtools-protocol/tot/Page#type-Frame */
```

Comment:
ah perhaps my question above is already answered then :)

this is somewhat curious, I would at least expect the root OOPIF entry to appear even if we don't get the OOPIF-child information. I expect this to be very relevant for ads in particular. Then again, I'm not really sure what additional information `frame` really conveys to the user. How is it used in the adspeed plugin?

Reply:
it certainly is relevant, but it doesn't seem like there's any way for us to access it, open to suggestions though if you have any thoughts.

> How is it used in the adspeed plugin?

It's used in the [paint timing audit](https://github.com/googleads/pub-ads-lighthouse-plugin/blob/master/lighthouse-plugin-ad-speed-insights/audits/first-ad-paint.js), to match the paint trace events with corresponding iframes